### ISSUE TITLE
YOLObile: Real-Time Object Detection on Mobile Devices via Compression-Compilation Co-Design

### ISSUE THREAD

**Issue by Studds-PM** [AlexeyAB/darknet#6736]
For Feature-request:
    * describe your feature as detailed as possible
YOLObile framework, a real-time object detection on mobile devices via compression-compilation co-design.

    * provide link to the paper and/or source code if it exist
[https://arxiv.org/abs/2009.05697](url)

[https://arxiv.org/pdf/2009.05697v1.pdf](url)

    * attach chart/table with comparison that shows improvement
![fps_map_coco](https://user-images.githubusercontent.com/69042118/94231983-b17dbd00-ff22-11ea-9c2c-bfa57bacd314.JPG)

![acc_speed_yolobile](https://user-images.githubusercontent.com/69042118/94231996-b6427100-ff22-11ea-9cac-c129f403b246.JPG)

![model_comparison](https://user-images.githubusercontent.com/69042118/94232035-c9554100-ff22-11ea-8c0c-d1bc35146f4c.JPG)

**Comment by wuzhenxin1989** [comment]
How to train my data

**Comment by Studds-PM** [comment]
@WongKinYiu The author has mentioned in his paper that he has trained the model using Pytorch but have not shared it publically yet. So either we need to implement by ourselves or wait for the public repo.

**Comment by 1343464520** [comment]
i also wana know how to use **yolobile** in darknet framework, so that i can implement it using OpenCV DNN. i am expecting that @AlexeyAB could realize it recently...

**Comment by DrFirestream** [comment]
The weights are released and tested here:
https://github.com/CoCoPIE-Pruning/CoCoPIE-ModelZoo/tree/master/YOLObile

Concerning the speed improvements, it seems the DNN framework  should be VERY good in handling sparse weights. Their approach is basically unstructured weights pruning, with a bit of structure on top, where they group the weights in blocks of 4 channels and 8 filters (x32). This structure allows them to improve parallelism but that's really hardware specific and hard to realize. That's impossible to do in pure pytorch. On the other hand up to 93% of reduction in weight size without real loss in accuracy is really tempting...

**Comment by AlexeyAB** [comment]
@DrFirestream 
Yes, it is very difficult to parallelize unstuctured sparsity.
While structured sparsity can be well parallelized, so it can be used on GPU/NPU: https://github.com/AlexeyAB/darknet/issues/6802
